utils/calendrical_calculations/src/chinese_based.rs

```diff
@@ -473,6 +473,65 @@ pub fn days_in_month<C: ChineseBased>(
     (result, next_new_moon)
 }
 
+/// Returns the last day of every month in the year, as well as a leap month index (1-indexed) if any
+/// In the case of no leap months, month 12 and 13 will have identical "last day"
+/// values
+pub fn month_structure_for_year<C: ChineseBased>(
+    new_year: RataDie,
+    next_new_year: RataDie,
+) -> ([RataDie; 13], Option<NonZeroU8>) {
+    let mut ret = [RataDie::new(0); 13];
+    ret[12] = next_new_year - 1; // last day of last month is the day before the next new year
+
+    let mut current_month_start = new_year;
+    let mut current_month_major_solar_term = major_solar_term_from_fixed::<C>(new_year);
```

Comment:
Nit: I prefer not introducing new code that uses the astronomy functions directly. Prefer to use abstractions we already have such as `days_in_month`.

Reply:
ok, not super comfortable reviewing astronomy code that isn't a direct port from lisp but thanks for writing the unit tests below